src/compiler/types.ts

```diff
@@ -169,6 +170,7 @@ module ts {
         ConstructSignature,
         IndexSignature,
         // Type
+        TypeGuardType,

```

Comment:
An alternative name that I think I like better is TypePredicate or PredicateType